node-commons/src/main/java/org/apache/iotdb/commons/client/ClientManagerProperty.java

```diff
@@ -0,0 +1,144 @@
+    // a node exceeds this number, newly returned clients will be released
+    private int maxIdleConnectionForEachNode = DefaultProperty.MAX_IDLE_CONNECTION_FOR_EACH_NODE;
+
+    // whether to use thrift compression
+    private boolean rpcThriftCompressionEnabled = DefaultProperty.RPC_THRIFT_COMPRESSED_ENABLED;
+    // socket timeout for thrift client
+    private int connectionTimeoutMs = DefaultProperty.CONNECTION_TIMEOUT_MS;
+    // number of selector threads for asynchronous thrift client
+    private int selectorNumOfAsyncClientPool = DefaultProperty.SELECTOR_NUM_OF_ASYNC_CLIENT_POOL;
+
+    public Builder<V> setWaitClientTimeoutMS(long waitClientTimeoutMS) {
+      this.waitClientTimeoutMS = waitClientTimeoutMS;
+      return this;
+    }
+
+    public Builder<V> setMaxTotalConnectionForEachNode(int maxTotalConnectionForEachNode) {
+      this.maxTotalConnectionForEachNode = maxTotalConnectionForEachNode;
+      return this;
+    }
+
+    public Builder<V> setMaxIdleConnectionForEachNode(int maxIdleConnectionForEachNode) {
+      this.maxIdleConnectionForEachNode = maxIdleConnectionForEachNode;
+      return this;
+    }
+
+    public Builder<V> setRpcThriftCompressionEnabled(boolean rpcThriftCompressionEnabled) {
+      this.rpcThriftCompressionEnabled = rpcThriftCompressionEnabled;
+      return this;
+    }
+
+    public Builder<V> setConnectionTimeoutMs(int connectionTimeoutMs) {
+      this.connectionTimeoutMs = connectionTimeoutMs;
+      return this;
+    }
+
+    public Builder<V> setSelectorNumOfAsyncClientPool(int selectorNumOfAsyncClientPool) {
+      this.selectorNumOfAsyncClientPool = selectorNumOfAsyncClientPool;
+      return this;
+    }
+
+    public ClientManagerProperty<V> build() {
+      GenericKeyedObjectPoolConfig<V> poolConfig = new GenericKeyedObjectPoolConfig<>();
+      poolConfig.setMaxTotalPerKey(maxTotalConnectionForEachNode);
+      poolConfig.setMaxIdlePerKey(maxIdleConnectionForEachNode);
+      poolConfig.setMaxWait(Duration.ofMillis(waitClientTimeoutMS));
+      poolConfig.setTestOnReturn(true);
+      poolConfig.setTestOnBorrow(true);
+      return new ClientManagerProperty<>(
+          poolConfig,
+          rpcThriftCompressionEnabled
+              ? new TCompactProtocol.Factory()
+              : new TBinaryProtocol.Factory(),
+          connectionTimeoutMs,
+          selectorNumOfAsyncClientPool);
+    }
+  }
+
+  public static class DefaultProperty {
+
+    private DefaultProperty() {}
+
+    // pool config
+    public static final long WAIT_CLIENT_TIMEOUT_MS = 10_000;
+    public static final int MAX_TOTAL_CONNECTION_FOR_EACH_NODE = 100;
+    public static final int MAX_IDLE_CONNECTION_FOR_EACH_NODE = 100;
+
+    // thrift client config
+    public static final boolean RPC_THRIFT_COMPRESSED_ENABLED = false;
+    public static final int CONNECTION_TIMEOUT_MS = 20_000;
+    public static final int SELECTOR_NUM_OF_ASYNC_CLIENT_POOL = 1;
+  }
```

Comment:
I wonder why shouldn't we use the .properties file to maintain these parameters?

Reply:
Actually, in the ConfigNode and Server modules we use the default values in the configuration file, but in the node-commons module we also need to give a default parameter value, although in most cases it will be overwritten